src/Razor/Microsoft.NET.Sdk.Razor/src/CreateBlazorTrimmerRootDescriptorFile.cs

```diff
@@ -21,16 +23,27 @@ public class CreateBlazorTrimmerRootDescriptorFile : Task
 
         public override bool Execute()
         {
-            using var fileStream = File.Create(TrimmerFile.ItemSpec);
+            var rootDescriptor = CreateRootDescriptorContents();
```

Comment:
Adding the extra package file uncovered an issue with publish incrementalism. We were always writing a new trimmer descriptor which would cause the linker to be non-incremental. The task was updated to use the same pattern as `WriteLinesToFile.WriteOnlyWhenDifferent`